crates/league-toolkit/src/core/wad/builder.rs

```diff
@@ -0,0 +1,197 @@
+use std::io::{self, BufReader, BufWriter, Cursor, Read, Seek, SeekFrom, Write};
+
+use byteorder::{WriteBytesExt, LE};
+use flate2::read::GzEncoder;
+use io_ext::measure;
+use itertools::Itertools;
+use xxhash_rust::{xxh3, xxh64};
+
+use crate::league_file::LeagueFileKind;
+
+use super::{WadChunk, WadChunkCompression, WadError};
+
+#[derive(Debug, thiserror::Error)]
+pub enum WadBuilderError {
+    #[error("wad error")]
+    WadError(#[from] WadError),
+
+    #[error("io error")]
+    IoError(#[from] io::Error),
+
+    #[error("unsupported compression type: {0}")]
+    UnsupportedCompressionType(WadChunkCompression),
+}
+
+/// Implements a builder interface for creating WAD files.
+///
+/// # Examples
+/// ```
+/// # use league_toolkit::core::wad::*;
+/// #
+/// let builder = WadBuilder::default();
+/// builder.with_chunk(WadChunkBuilder::default().with_path("path/to/chunk"));
+/// builder.build_to_writer(File::create("output.wad").unwrap());
```

Comment:
```suggestion
/// builder.build_to_writer(File::create("output.wad")?);
/// # Ok::<(),std::io::Error>(())
```
haven't actually checked the ok type there is correct but minor improvement to readability :3